File: app/src/main/java/theredspy15/ltecleanerfoss/controllers/PromptActivity.java
/*
 * Copyright 2020 Hunter J Drum
 */

package theredspy15.ltecleanerfoss.controllers;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

import theredspy15.ltecleanerfoss.R;

public class PromptActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_prompt);

        Button button = findViewById(R.id.button1);
        button.setOnClickListener(view -> {
            resultLauncher.launch(new Intent(android.provider.Settings.ACTION_SETTINGS));
            System.exit(0);
        });
    }

    //Instead of onActivityResult() method use this one
    ActivityResultLauncher<Intent> resultLauncher = registerForActivityResult(
            new ActivityResultContracts.StartActivityForResult(),
            result -> {
                if (result.getResultCode() == Activity.RESULT_OK) {
                    // Here, no request code
                    Intent data = result.getData();
                }
            });
}

File: app/src/main/java/theredspy15/ltecleanerfoss/controllers/PromptActivity.kt
/*
 * Copyright 2020 Hunter J Drum
 */
package theredspy15.ltecleanerfoss.controllers

import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.widget.Button
import androidx.activity.result.ActivityResult
import androidx.activity.result.contract.ActivityResultContracts.StartActivityForResult
import androidx.appcompat.app.AppCompatActivity
import theredspy15.ltecleanerfoss.R
import kotlin.system.exitProcess

class PromptActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_prompt)
        val button = findViewById<Button>(R.id.button1)
        button.setOnClickListener {
            resultLauncher.launch(Intent(Settings.ACTION_SETTINGS))
            exitProcess(0)
        }
    }

    //Instead of onActivityResult() method use this one
    private var resultLauncher = registerForActivityResult(
        StartActivityForResult()
    ) { result: ActivityResult ->
        if (result.resultCode == RESULT_OK) {
            // Here, no request code
            val data = result.data
        }
    }
}